### PR TITLE
Heimdal checksum fixes

### PR DESCRIPTION
**Pull request by metze-samba**
Please merge these fixes which are already in Samba's copy of heimdal.

Thanks!

### CODE REVIEW

File: lib/krb5/crypto.c [hunk 1]
```diff
@@ -316,6 +316,24 @@ get_checksum_key(krb5_context context,
...
+			       N_("Checksum type %s is keyed but no "
+			          "crypto context (key) was passed in", ""),
+			       ct->name);
+	return KRB5_BAD_ENCTYPE;
+    }
+    kct = crypto->et->keyed_checksum;
+    if (kct == NULL || kct->type != ct->type) {
+	krb5_set_error_message(context, KRB5_BAD_ENCTYPE,
+			       N_("Checksum type %s is keyed, but "
+			          "the key type %s passed didnt have that checksum "
```
Comment (vdukhovni) [inline lib/krb5/crypto.c]: Nit.  EIther say "didn't" with an apostrophe, or else "did not".

**Comment by vdukhovni** [comment]
Overall, looks good to me.
@nicowilliams: care to look?

**Comment by nicowilliams** [comment]
I'll take a look.

**Comment by nicowilliams** [comment]
Thanks!  Sorry it took so long for me to review this!

**Pull request merged**